rfcs/s3-implement-virtual-link.md

```diff
@@ -0,0 +1,56 @@
+- Author: abyss-w <mad.hatter@foxmailcom>
+- Start Date: 2021-08-05
+- RFC PR: []()
+- Tracking Issue: [beyondstorage/go-service-s3#144](https://github.com/beyondstorage/go-service-s3/issues/144)
+
+# GSP-xxx: S3 Implementation of Virtual Link
+
+## Background
+
+Like the one presented in [GSP-86 Add Create Link Operation](https://github.com/beyondstorage/go-storage/blob/master/docs/rfcs/86-add-create-link-operation.md), s3 has no native support for symlink. We can use [x-amz-website-redirect-location](https://docs.aws.amazon.com/AmazonS3/latest/userguide/how-to-page-redirect.html) redirect pages but only works for a website. 

```

Comment:
Since `x-amz-website-redirect-location` only works for website and it's also mentioned in https://github.com/beyondstorage/go-service-s3/pull/175#discussion_r686470881，I‘m confuse about the universality for the interface. Or we can support or allow implementations for specific scenarios.

Reply:
> If your Amazon S3 bucket is configured for static website hosting, you can configure redirects for your bucket or the objects in it. You have the following options for configuring redirects. 

Our cases don't meet this scenario.

Add a user customed metadata is OK to me. But the key should be prefixed with `bs-`, so we can avoid the conflicts.

Finally, we will get key like `x-amz-bs-link-target`.

Reply:
ping @abyss-w for a revisit.

Reply:
Got it.

Reply:
> Add a user customed metadata is OK to me. 

Then we need to reconsider whether other services that support user-defined metadata can implement(simulate) it.